source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -526,17 +538,22 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
   if (ci.get()) {
     cluster_metadata = &ci->metadata();
   }
-
-  // start transforming!
-  TransformerInstance instance(header_map, request_headers, get_body,
-                               extractions, json_body, environ_,
-                               cluster_metadata, rng_);
-
+  tls_->set([header_map=&header_map, request_headers, get_body, extractions=&extractions, json_body=&json_body, environ=&environ_, cluster_metadata, rng=&rng_](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
```

Comment:
you can't call `set()` from `transform()` as `set` will set the object on all worker threads.
call `set()` from the ctor (you only need to call it one time).
Here you can call `getTyped` and update the existing object

Reply:
I changed to have a `set()` call construct an empty object, then it is populated with request context by getting a reference to the thread-local object with `getTyped`. Is this more in line with how this should be used?